### PR TITLE
[ci skip] Update development_dependencies_install document for nodesource setup

### PR DESCRIPTION
**Pull request by m11o**
### Motivation / Background

ref: #49900 

According to [nodesource/distributions](https://github.com/nodesource/distributions),

> The installation scripts `setup_XX.x` are no longer supported and are not needed anymore.

So, This Pull Request has been created because of updating to new installation method for nodejs by using  [nodesource/distributions](https://github.com/nodesource/distributions).

### Detail

This Pull Request removes nodesource script (`setup_XX.x`) and updates nodejs installation method for `Ubuntu` and `Fedora or CentOS` in development_dependencies_install.md.

### Additional information

- [Installation Instructions for Debian and Ubuntu based distributions](https://github.com/nodesource/distributions?tab=readme-ov-file#installation-instructions)
- [Installation Instructions for Enterprise Linux Based Distributions](https://github.com/nodesource/distributions?tab=readme-ov-file#installation-instructions-1)

### Checklist

Before submitting the PR make sure the following are checked:

* [x] This Pull Request is related to one change. Changes that are unrelated should be opened in separate PRs.
* [x] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`
* [x] Tests are added or updated if you fix a bug or add a feature.
* [x] CHANGELOG files are updated for the changed libraries if there is a behavior change or additional feature. Minor bug fixes and documentation changes should not be included.

### CODE REVIEW

File: guides/source/development_dependencies_install.md [hunk 1]
```diff
@@ -123,8 +123,13 @@ $ sudo apt-get install sqlite3 libsqlite3-dev mysql-server libmysqlclient-dev po
 
 # Install Yarn
 # Use this command if you do not have Node.js installed
-$ curl --fail --silent --show-error --location https://deb.nodesource.com/setup_18.x | sudo -E bash -
+# ref: https://github.com/nodesource/distributions#installation-instructions
+$ sudo mkdir -p /etc/apt/keyrings
+$ curl --fail --silent --show-error --location https://deb.nodesource.com/gpgkey/nodesource-repo.gpg.key | sudo gpg --dearmor -o /etc/apt/keyrings/nodesource.gpg
+$ echo "deb [signed-by=/etc/apt/keyrings/nodesource.gpg] https://deb.nodesource.com/node_18.x nodistro main" | sudo tee /etc/apt/sources.list.d/nodesource.list
```
Comment (vipulnsward) [inline guides/source/development_dependencies_install.md:129]: Lets updated to 20 while we are here, current lts- https://nodejs.org/en

Comment (m11o) [reply guides/source/development_dependencies_install.md]: Thank you for your review.
I fixed it in d7396bfc20c3d20b5376fb2ece86c0b9694f76ca

**Comment by zzak** [comment]
@m11o Thanks for your patch! Could you please [squash your commits](https://edgeguides.rubyonrails.org/contributing_to_ruby_on_rails.html#squashing-commits)? 🙇

**Comment by m11o** [comment]
@zzak 
Of course! Thank you for telling me that.
I fixed that in 9a1d027c974609d26b181b9c4faec21574f49eb8.

**Review by zzak — APPROVED** [review]

**Review by akhilgkrishnan — APPROVED** [review]

**Pull request merged**